pkg/deploy/generator/resources_rp.go

```diff
@@ -410,6 +411,18 @@ func (g *generator) rpVMSS() *arm.Resource {
 		)
 	}
 
+	// convert array variables to string using ARM string() function to be passed via customScript later
+	for _, variable := range []string{
+		"miseValidAudiences",
+		"miseValidAppIDs",
+	} {
+		parts = append(parts,
+			fmt.Sprintf("'%s=$(base64 -d <<<'''", strings.ToUpper(variable)),
+			fmt.Sprintf("base64(string(parameters('%s')))", variable),
```

Comment:
If this variable is manually input, there's a chance a space character is added. To ensure there is no extra character, I think it's best to trim this variable first.

Reply:
These variables are not being input manually but rather are being fetched from the configurations in RP-Config repository.